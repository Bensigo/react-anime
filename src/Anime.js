// @flow

import React from 'react';
import anime from 'animejs';

class Anime extends React.Component {
  constructor (props) {
    super(props);
  }

  componentDidMount () {

  }

  componentWillUnmount () {

  }
  
  render () {
    return null; 
  }
}

export default Anime;
