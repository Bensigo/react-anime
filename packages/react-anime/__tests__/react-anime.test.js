import React from 'react';
import { renderToString } from 'react-dom';
import renderer from 'react-test-renderer';
import Anime from '../src/react-anime';

//snapshot test
it('2 children delayed fade in', () => {
  const tree = renderer.create(
    <Anime
      easing="easeOutElastic"
      loop={true}
      duration={1000}
      direction="alternate"
      loop={true}
      delay={(el, index) => index * 240}
    >
      <p>1</p>
      <p>2</p>
    </Anime>,
    {
      createNodeMock: element => ({ nodeType: true }),
    },
  );

  expect(tree.toJSON()).toMatchSnapshot();
});