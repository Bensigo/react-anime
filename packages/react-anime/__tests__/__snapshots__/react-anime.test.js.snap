// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anime single child Snapshot 1`] = `
<g>
  <p>
    1
  </p>
  <p>
    2
  </p>
</g>
`;
